File: mobile/lib/features/condition_travail/domain/entities/programme_chauffeur.dart
import '../../../chauffeur/domain/entities/chauffeur.dart';

class ProgrammeChauffeur {
  final int? id;
  final Chauffeur chauffeur;
  final int ordreAlternance;
  final int? ordreJourSalaire;
  final DateTime? dateService;

  const ProgrammeChauffeur({
    this.id,
    required this.chauffeur,
    required this.ordreAlternance,
    this.ordreJourSalaire,
    this.dateService,
  });

  int get chauffeurId => chauffeur.id ?? 0;

  String get nomComplet => chauffeur.fullName;

  ProgrammeChauffeur copyWith({
    int? id,
    Chauffeur? chauffeur,
    int? ordreAlternance,
    int? ordreJourSalaire,
    DateTime? dateService,
  }) {
    return ProgrammeChauffeur(
      id: id ?? this.id,
      chauffeur: chauffeur ?? this.chauffeur,
      ordreAlternance: ordreAlternance ?? this.ordreAlternance,
      ordreJourSalaire: ordreJourSalaire ?? this.ordreJourSalaire,
      dateService: dateService ?? this.dateService,
    );
  }
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/programmeTravail/ProgrammeChauffeur.java
package com.tmk.vtcmanager.application.domain.programmeTravail;

import com.tmk.vtcmanager.application.domain.chauffeur.Chauffeur;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProgrammeChauffeur {

    private Long id;
    private Chauffeur chauffeur;
    private Integer ordreAlternance;
    private Integer ordreJourSalaire;
    private LocalDate dateService;

    public Long getChauffeurId() {
        return chauffeur != null ? chauffeur.getId() : null;
    }

    public String getNomComplet() {
        return chauffeur != null ? chauffeur.getFullName() : "";
    }
}
